uuid/Data/UUID/TH.hs

```diff
@@ -0,0 +1,30 @@
+{-# LANGUAGE TemplateHaskell #-}
```

Comment:
Why do we need this extension enabled? (NB: if we enable this, `uuid` stops working on TH-less GHCs)

Reply:
Ok, in that case you should use `TemplateHaskellQuotes` for the GHC versions that support it, and `TemplateHaskell` for those that don't; this ensures that the package will keeping working with TH-less GHCs for recent GHC versions at least

Reply:
Do I use the CPP and a version check for GHC > 8.0 to make the decision which to use, or is there some better way of doing it?